internal/renderers/skia/opengl_surface.rs

```diff
@@ -79,69 +94,86 @@ impl super::Surface for OpenGLSurface {
         callback(api)
     }
 
-    fn with_active_surface(&self, callback: impl FnOnce()) {
-        self.ensure_context_current();
+    fn with_active_surface(&self, callback: impl FnOnce()) -> Result<(), PlatformError> {
+        self.ensure_context_current()?;
         callback();
+        Ok(())
     }
 
     fn render(
         &self,
         size: PhysicalWindowSize,
         callback: impl FnOnce(&mut skia_safe::Canvas, &mut skia_safe::gpu::DirectContext),
-    ) {
-        let width = size.width;
-        let height = size.height;
-
-        self.ensure_context_current();
+    ) -> Result<(), PlatformError> {
+        self.ensure_context_current()?;
 
         let current_context = &self.glutin_context;
 
         let gr_context = &mut self.gr_context.borrow_mut();
 
         let mut surface = self.surface.borrow_mut();
-        if width != surface.width() as u32 || height != surface.height() as u32 {
+        if size.width.try_into().ok() != Some(surface.width())
```

Comment:
since create_internal_surface is going to fail if with or height = 0,  we probably shouldn't call it at all.

Reply:
Good point!